### PR TITLE
Update database software requirement

### PR DESCRIPTION
**Pull request by james-mm**
Added known postgres limitation

### CODE REVIEW

**Comment by lfbrock** [comment]
Can we add a note about MySQL limitations as well?

**Comment by james-mm** [comment]
@lindy65 Good idea, yes, indicating that postgres should be an enterprise feature so it does not come as a surprise to those who are looking to enable it.

**Comment by james-mm** [comment]
@lfbrock I don't know what the MySQL limitations are. Feel free to update, or let me know and I can make the update.  Same with updating the additional postrgres limitation on dashes

**Comment by esethna** [comment]
@james-mm we have some notes from the search doc that might be helpful: 

https://docs.mattermost.com/help/getting-started/searching.html?highlight=searching#technical-notes

image

**Comment by james-mm** [comment]
Updates done

File: source/install/requirements.rst [hunk 1]
```diff
@@ -70,6 +70,17 @@ Database Software
 
 Deployments requiring searching in Chinese, Japanese and Korean languages require MySQL 5.7.6+ and the configuration of `ngram Full-Text parser <https://dev.mysql.com/doc/refman/5.7/en/fulltext-search-ngram.html>`__. For searching two characters, you will also need to set ``ft_min_word_len`` and ``innodb_ft_min_token_size`` to ``2`` and restart MySQL. See `CJK discussion <https://github.com/mattermost/mattermost-server/issues/2033#issuecomment-183872616>`__ for details.
 
+Notes on PostgreSQL:
+
+- Email addresses do not return results.
+- Hashtags or recent mentions of usernames containing a dash do not return search results.
+- Terms containing a dash return incorrect results as dashes are ignored in the search engine.
+- If any of the above is an issue, you can either enable the ElasticSearch (E20) feature or install MySQL instead.
```
Comment (esethna) [inline source/install/requirements.rst]: It's actually, "Elasticsearch", no capital S

File: source/install/requirements.rst [hunk 2]
```diff
@@ -70,6 +70,17 @@ Database Software
 
 Deployments requiring searching in Chinese, Japanese and Korean languages require MySQL 5.7.6+ and the configuration of `ngram Full-Text parser <https://dev.mysql.com/doc/refman/5.7/en/fulltext-search-ngram.html>`__. For searching two characters, you will also need to set ``ft_min_word_len`` and ``innodb_ft_min_token_size`` to ``2`` and restart MySQL. See `CJK discussion <https://github.com/mattermost/mattermost-server/issues/2033#issuecomment-183872616>`__ for details.
 
+Notes on PostgreSQL:
```
Comment (esethna) [inline source/install/requirements.rst]: Propose "Search limitations on PostgreSQL:"

File: source/install/requirements.rst [hunk 3]
```diff
@@ -70,6 +70,17 @@ Database Software
...
+
+- Email addresses do not return results.
+- Hashtags or recent mentions of usernames containing a dash do not return search results.
+- Terms containing a dash return incorrect results as dashes are ignored in the search engine.
+- If any of the above is an issue, you can either enable the ElasticSearch (E20) feature or install MySQL instead.
+
+Notes on MySQL:
```
Comment (esethna) [inline source/install/requirements.rst]: Propose "Search limitations on MySQL:"

File: source/install/requirements.rst [hunk 4]
```diff
@@ -70,6 +70,17 @@ Database Software
 
 Deployments requiring searching in Chinese, Japanese and Korean languages require MySQL 5.7.6+ and the configuration of `ngram Full-Text parser <https://dev.mysql.com/doc/refman/5.7/en/fulltext-search-ngram.html>`__. For searching two characters, you will also need to set ``ft_min_word_len`` and ``innodb_ft_min_token_size`` to ``2`` and restart MySQL. See `CJK discussion <https://github.com/mattermost/mattermost-server/issues/2033#issuecomment-183872616>`__ for details.
 
+Notes on PostgreSQL:
+
+- Email addresses do not return results.
+- Hashtags or recent mentions of usernames containing a dash do not return search results.
+- Terms containing a dash return incorrect results as dashes are ignored in the search engine.
```
Comment (esethna) [inline source/install/requirements.rst]: Propose "search engine" > "search query"

File: source/install/requirements.rst (hunk 1 above)
Comment (esethna) [reply source/install/requirements.rst]: Propose adding a link to the elasticsearch docs from the word

File: source/install/requirements.rst (hunk 2 above)
Comment (james-mm) [reply source/install/requirements.rst]: Done

File: source/install/requirements.rst (hunk 4 above)
Comment (james-mm) [reply source/install/requirements.rst]: Done

File: source/install/requirements.rst (hunk 1 above)
Comment (james-mm) [reply source/install/requirements.rst]: done

File: source/install/requirements.rst (hunk 3 above)
Comment (james-mm) [reply source/install/requirements.rst]: Done

**Comment by james-mm** [comment]
Update done per @esethna 's proposed changes

**Pull request merged**